bundles/org.eclipse.swt/Eclipse SWT Drag and Drop/gtk/org/eclipse/swt/dnd/TextTransfer.java

```diff
@@ -107,8 +111,29 @@ public void javaToNative (Object object, TransferData transferData) {
 		transferData.pValue = string_target;
 		transferData.result = 1;
 	}
+    if (transferData.type == TEXT_PLAIN_UTF8_ID) {
+        // Convert the text into RFC-1341 format
+        byte[] rfc1341Data = encodeTextAsRFC1341(string);
+        transferData.format = 8; // Format for UTF-8
+        transferData.length = rfc1341Data.length;
+        transferData.pValue = OS.g_malloc(rfc1341Data.length);
+        if (transferData.pValue != 0) {
+            C.memmove(transferData.pValue, rfc1341Data, rfc1341Data.length);
+            transferData.result = 1;
+        }
+    }
+}
+
+// New method to encode text as RFC-1341
+private byte[] encodeTextAsRFC1341(String text) {
+    // Implement encoding logic here, e.g., adding MIME headers and encoding text
+    // This is a simplified example; actual encoding depends on RFC-1341 standards
+//    String rfc1341Text = "Content-Type: " + TEXTPLAINUTF8 + "\r\n\r\n" + text;
+    String rfc1341Text = text;
+    return rfc1341Text.getBytes(StandardCharsets.UTF_8);
```

Comment:
Please see https://github.com/eclipse-platform/eclipse.platform.swt/pull/875#discussion_r1389220864

I have my doubts what this method does.

Reply:
This method converts strings to-utf8
https://stackoverflow.com/questions/88838/how-to-convert-strings-to-and-from-utf8-byte-arrays-in-java

[How to convert Strings to and from UTF8 byte arrays in Java](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwilgt6zubmCAxWqRvEDHU0MBU8QFnoECA0QAQ&url=https%3A%2F%2Fstackoverflow.com%2Fquestions%2F88838%2Fhow-to-convert-strings-to-and-from-utf8-byte-arrays-in-java&usg=AOvVaw3Gw2s2O_59SDt8PUprFZsS&opi=89978449)

Reply:
Maybe the method should be named `convertToUTF8`, as RFC 1341 is the standard that talks about the formatting if "MIME types" (i.e. the `maintype/subtype;options` format), and isn't actually about UTF strings. Also, the code of the method is pretty simple - effectively a short one liner - so maybe it can just be used directly in the one place it is being used.